### PR TITLE
Fix data-gatherers config loading

### PR DESCRIPTION
**Pull request by j-fuentes**
This is needed so we load the config of the particular data-gatherers.

Signed-off-by: Jose Fuentes <jsfuentescastillo@gmail.com>

### CODE REVIEW

File: pkg/agent/config.go [hunk 1]
```diff
@@ -25,10 +31,76 @@ type Endpoint struct {
...
+		}{}
+		err = unmarshal(&cfg)
+		dg.Config = cfg.Config
+	// dummy dataGatherer is just used for testing
+	case "dummy":
+		cfg := struct {
+			Config *dummyConfig `yaml:"config"`
+		}{}
+		err = unmarshal(&cfg)
+		dg.Config = cfg.Config
```
Comment (j-fuentes) [inline pkg/agent/config.go]: I am sure there is a better way of doing this with `reflect`, but I am too tired at the moment.

I will revisit this tomorrow.

Comment (j-fuentes) [reply pkg/agent/config.go]: I have reworked this part. Hopefully, it is better now.

File: pkg/agent/config.go [hunk 2]
```diff
@@ -25,10 +31,72 @@ type Endpoint struct {
...
+	err = yaml.Unmarshal(bb, config)
+	if err != nil {
+		return nil
+	}
+
+	return nil
+}
+
+// UnmarshalYAML unmarshals a dataGatherer resolving the type according to Kind.
+func (dg *dataGatherer) UnmarshalYAML(unmarshal func(interface{}) error) error {
```
Comment (charlieegan3) [inline pkg/agent/config.go]: This is called implicitly when loading yaml data into a dg right?

File: pkg/agent/config.go (hunk 1 above)
Comment (charlieegan3) [reply pkg/agent/config.go]: The new version looks :+1:

**Comment by charlieegan3** [comment]
I'm going to lgtm - but would be nice to confirm this question here: https://github.com/jetstack/preflight/pull/119#discussion_r383210502

/lgtm
/approve

File: pkg/agent/config.go (hunk 2 above)
Comment (j-fuentes) [reply pkg/agent/config.go]: Exactly: https://godoc.org/gopkg.in/yaml.v2#Unmarshaler

Comment (charlieegan3) [reply pkg/agent/config.go]: :+1: thanks.

**Pull request merged**